src/routers/users.js

```diff
@@ -0,0 +1,120 @@
+'use strict';
+
+const express = require('express');
+const { getNewId } = require('../helpers');
+const router = express.Router();
+
+let users = [];
+
+function setInitialUsers() {
+  users = [];
+}
+
+const filterUsers = (userId) => (
+  users.filter(user => Number(userId) !== user.id)
+);
+
+const getUserById = (userId) => {
+  return users.find(user => user.id === Number(userId));
+};
+
+router.get('/', (req, res) => {
+  res.status(200);
+  res.json(users);
+});
+
+router.get('/:userId', (req, res) => {
+  const { userId } = req.params;
+
+  const userById = getUserById(userId);
+
+  if (!userById) {
+    res.status(404).send({ message: 'Expected entity doesn/t exist.' });
+
+    return;
+  }
+  res.status(200);
+  res.json(userById);
+});
+
+router.post('/', (req, res) => {
+  const { name } = req.body;
+
+  if (!name) {
+    res.status(400).send({ message: 'Required parameter is not passed' });
+
+    return;
+  }
+
+  const newUser = {
+    id: getNewId(users),
+    name,
+  };
+
+  users.push(newUser);
+  res.status(201);
+  res.json(newUser);
+});
```

Comment:
Let's create 3 layouts. Routes, Controllers, Services.

For example:

**routes/users.js**

`router.post('/', userController.createUser);`

**controller/users.js**

```
const createUser = (req, res) => {
  const { name } = req.body;

  if (!name) {
    res.status(400);
    res.send('User is not added');

    return;
  }
  const newUser = userService.createUser(name);

  res.statusCode = 201;
  res.json(newUser);
}
```

And in service, you need to add the logic of interaction with the database.
Same in all other cases